HGCalAnalysis/plugins/HGCalAnalysis.cc

```diff
@@ -153,15 +153,15 @@ HGCalAnalysis::HGCalAnalysis(const edm::ParameterSet& iConfig) :
   if(!readOfficialReco) {
     _vtx = consumes<std::vector<TrackingVertex> >(edm::InputTag("mix","MergedTrackTruth"));
     _part = consumes<std::vector<TrackingParticle> >(edm::InputTag("mix","MergedTrackTruth"));
-    if (!readCaloParticles) {
-      _caloParticles = consumes<std::vector<CaloParticle> >(edm::InputTag("mix","MergedCaloTruth"));
-    }
   }
   else {
     _hev = consumes<edm::HepMCProduct>(edm::InputTag("generatorSmeared") );
     _simTracks = consumes<std::vector<SimTrack> >(edm::InputTag("g4SimHits"));
     _simVertices = consumes<std::vector<SimVertex> >(edm::InputTag("g4SimHits"));
   }
+  if (readCaloParticles) {
```

Comment:
~~While_ this line is correct, there's another problem, since further below there's no check if `readOfficialReco` is false, so the code is likely to crash when running on RelVals with `readCaloParticles` is true~~
OK, I fixed it when resolving the conflict, will merge.